tests/base.py

```diff
@@ -0,0 +1,38 @@
+import distutils
+import inspect
+import os.path
+import subprocess
+import types
+try:
+  import unittest2 as unittest
+except ImportError:
+  import unittest
+
+
+class PyangBindTestCase(unittest.TestCase):
+  yang_files = None
+
+  @classmethod
+  def setUpClass(cls):
+    if cls.yang_files is None:
+      raise ValueError("cls.yang_files must be set")
+    pyang_path = distutils.spawn.find_executable('pyang')
+    if not pyang_path:
+      raise RuntimeError("Could not locate `pyang` executable.")
+    base_dir = os.path.dirname(os.path.dirname(__file__))
+    test_path = os.path.dirname(inspect.getfile(cls))
+    yang_files = [os.path.join(test_path, filename) for filename in cls.yang_files]
+    plugin_dir = os.path.join(base_dir, 'pyangbind', 'plugin')
+
+    pyang_cmd = "{} --plugindir {} -f pybind -p {} --use-extmethods --use-xpathhelper {}".format(
```

Comment:
We probably need a bool option here for the `split-class-dir` flag. These base tests just produce a single `bindings.py` but the others generate whole class hierarchies.